### PR TITLE
Update Build Script Help Content

### PR DESCRIPTION
**Pull request by ViktorHofer**
Add the new runtime/library configuration arguments and shuffle a few things around to be more consistent.

cc @wfurt

### CODE REVIEW

**Review by safern — APPROVED** [review]

**Comment by danmosemsft** [comment]
Can you please update the markdowns if needed? I think most contributors will go off those.

**Comment by ViktorHofer** [comment]
I just took a lot. A docs update shouldn't be needed :)

File: eng/build.ps1 [hunk 1]
```diff
@@ -14,21 +14,24 @@ Param(
...
-  Write-Host "  -verbosity <value>      MSBuild verbosity: q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic] (short: -v)"
-  Write-Host "  -binaryLog              Output binary log (short: -bl)"
-  Write-Host "  -help                   Print help and exit (short: -h)"
+  Write-Host "  -subset                   Build a subset, print available subsets with -subset help"
+  Write-Host "  -subsetCategory           Build a subsetCategory, print available subsetCategories with -subset help"
+  Write-Host "  -vs                       Open the solution with VS for Test Explorer support. Path or solution name (ie -vs Microsoft.CSharp)"
+  Write-Host "  -os                       Build operating system: Windows_NT or Unix"
+  Write-Host "  -arch                     Build platform: x86, x64, arm or arm64"
+  Write-Host "  -configuration            Build configuration: Debug, Release or [CoreCLR]Checked (short: -c)"
+  Write-Host "  -runtimeConfiguration     Runtime build configuration: Debug, Checked or Release"
```
Comment (jaredpar) [inline eng/build.ps1]: Does this control both runtimes or just CoreCLR?

File: eng/build.ps1 [hunk 2]
```diff
@@ -14,21 +14,24 @@ Param(
...
-  Write-Host "  -configuration <value>  Build configuration: Debug or Release (short: -c)"
-  Write-Host "  -verbosity <value>      MSBuild verbosity: q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic] (short: -v)"
-  Write-Host "  -binaryLog              Output binary log (short: -bl)"
-  Write-Host "  -help                   Print help and exit (short: -h)"
+  Write-Host "  -subset                   Build a subset, print available subsets with -subset help"
+  Write-Host "  -subsetCategory           Build a subsetCategory, print available subsetCategories with -subset help"
+  Write-Host "  -vs                       Open the solution with VS for Test Explorer support. Path or solution name (ie -vs Microsoft.CSharp)"
+  Write-Host "  -os                       Build operating system: Windows_NT or Unix"
+  Write-Host "  -arch                     Build platform: x86, x64, arm or arm64"
+  Write-Host "  -configuration            Build configuration: Debug, Release or [CoreCLR]Checked (short: -c)"
```
Comment (jaredpar) [inline eng/build.ps1]: The OCD in me wants me to point out that some of this change uses "Debug, Release, Checked" and others use "Debug, Checked, Release". This is an ignorable comment :smile:

File: eng/build.sh [hunk 3]
```diff
@@ -21,8 +21,10 @@ usage()
   echo "  --subsetCategory           Build a subsetCategory, print available subsetCategories with -subset help"
   echo "  --os                       Build operating system: Windows_NT or Unix"
   echo "  --arch                     Build platform: x86, x64, arm or arm64"
-  echo "  --configuration <value>    Build configuration: Debug or Release (short: -c)"
-  echo "  --verbosity <value>        MSBuild verbosity: q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic] (short: -v)"
+  echo "  --configuration            Build configuration: Debug, Release or [CoreCLR]checked (short: -c)"
```
Comment (jaredpar) [inline eng/build.sh]: The PS version has `[CoreCLR]Checked` (note the C in checked is upper case)

**Review by jaredpar — APPROVED** [review]

File: eng/build.ps1 (hunk 1 above)
Comment (ViktorHofer) [reply eng/build.ps1]: yes.

File: eng/build.ps1 (hunk 2 above)
Comment (danmosemsft) [inline eng/build.ps1]: Does `[CoreCLR]Checked` mean I can choose either `Checked` or `CoreCLRChecked`? What's the difference? maybe clarify?

Comment (danmosemsft) [reply eng/build.ps1]: I assume it means only applies to CoreCLR, in which case `... Debug, Release, or (if CoreCLR) Checked` maybe

**Review by danmoseley — APPROVED** [review]

File: eng/build.ps1 (hunk 2 above)
Comment (ViktorHofer) [reply eng/build.ps1]: yeah I wasn't sure how to express that with the least amount of characters 😁

File: eng/build.sh [hunk 4]
```diff
@@ -21,8 +21,10 @@ usage()
   echo "  --subsetCategory           Build a subsetCategory, print available subsetCategories with -subset help"
   echo "  --os                       Build operating system: Windows_NT or Unix"
   echo "  --arch                     Build platform: x86, x64, arm or arm64"
-  echo "  --configuration <value>    Build configuration: Debug or Release (short: -c)"
-  echo "  --verbosity <value>        MSBuild verbosity: q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic] (short: -v)"
+  echo "  --configuration            Build configuration: Debug, Release or [CoreCLR]Checked (short: -c)"
+  echo "  --runtimeconfiguration     Runtime build configuration: Debug, Release or [CoreCLR]Checked"
+  echo "  --librariesconfiguration   Libraries build configuration: Debug or Release"
```
Comment (dagood) [inline eng/build.sh]: (Nit) Capitalize `c` in the above two lines to match ps `librariesConfiguration` and earlier arg docs in this file like `--subsetCategory`.

**Pull request merged**